dev/benchmarks/platform_channels_benchmarks/lib/main.dart

```diff
@@ -0,0 +1,148 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+import 'dart:typed_data';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:microbenchmarks/common.dart';
+
+List<Object> _makeTestBuffer(int size) {
+  final List<Object> answer = <Object>[];
+  for (int i = 0; i < size; ++i) {
+    switch (i % 9) {
+      case 0:
+        answer.add(1);
+        break;
+      case 1:
+        answer.add(math.pow(2, 65));
+        break;
+      case 2:
+        answer.add(1234.0);
+        break;
+      case 3:
+        answer.add(null);
+        break;
+      case 4:
+        answer.add(<int>[1234]);
+        break;
+      case 5:
+        answer.add(<String, int>{'hello': 1234});
+        break;
+      case 6:
+        answer.add('this is a test');
+        break;
+      case 7:
+        answer.add(true);
+        break;
+      case 8:
+        answer.add(Uint8List(64));
+        break;
+    }
+  }
+  return answer;
+}
+
+Future<void> _runTests() async {
+  assert(false,
+      "Don't run benchmarks in checked mode! Use 'flutter run --release'.");
```

Comment:
I'm pretty sure this was copied from elsewhere, but if we really want to check for release mode I think we want to do
```suggestion
  if (!kReleaseMode)
    throw Exception("Must be run in release mode! Use 'flutter run --release'.");
```

The assert will not be run in profile mode, and checked mode is an antiquated/deprecated notion at this point.

Reply:
done (checked against kDebugMode)

Reply:
Don't we want this check so that we make sure CI is running these in release mode? 

These benchmarks just rely on stdout to get timing information, rather than the observatory.

Reply:
No, there is a comment somewhere else that the tests have to be run in profile mode to work on iOS.  Plus the only real problem case is Debug if people are running them locally.